### PR TITLE
Improved combat system

### PR DESCRIPTION
**Pull request by ginterma**
Added attack animation for wolf and added push force on meele attacks for both player and wolf. Did not add push force on fireball because it does not work properly for some reason.

### CODE REVIEW

**Comment by MellowJay** [comment]
[[GRAPHICS] Wolf Attack Animation](https://trello.com/c/SKLCW2Mn/121-graphics-wolf-attack-animation)

**Comment by MellowJay** [comment]
[[FEATURE] Enemy Attack Patterns](https://trello.com/c/Mg8ehybh/55-feature-enemy-attack-patterns)

File: Assets/Scripts/EnemyScritps/EnemyAI.cs [hunk 1]
```diff
@@ -35,16 +36,21 @@ private void Awake()
...
         InvokeRepeating("CheckDist", 0, 1f);
-        hitbox = transform.GetChild(0).GetComponent<BoxCollider2D>();
         boxCollider = GetComponent<BoxCollider2D>();
     }
 
    public override void FixedUpdate()
     {
+        pushDirection = Vector2.Lerp(pushDirection, Vector2.zero, pushRecoverySpeed);
+        rb.MovePosition(rb.position + pushDirection);
+        if (myAnimator.GetBool("Attacking") == true)
```
Comment (MellowJay) [inline Assets/Scripts/EnemyScritps/EnemyAI.cs:48]: GetBool() returns a bool value. The "== true" is unnecessary.

**Review by MellowJay — CHANGES_REQUESTED** [review]
Ok, the wolf's attack animations work. Nice.
I'll try to fix the issue with the fireball and the animation property issue.

**Comment by MellowJay** [comment]
image

**Review by MellowJay — APPROVED** [review]
The wolf can't be pushed while moving, but I guess this is okay and makes sense.

**Review by MellowJay — CHANGES_REQUESTED** [review]
I changed the messages a little bit and fixed some spelling mistakes:  
- I wrote the "Press Enter To Proceed" phrase for every God dialog box, because after moving around I didn't know what to press to continue.
- I changed the robot talking about the fireball - it now mentions that pressing the f key casts a fireball, this is accurate on the dev branch, because spacebar performs a dash.  

My feedback on the tutorial experience overall:
- I think the dialog box can be positioned a little bit higher.
- There is a small dark line underneath the robot icon.
- The fire sword, dash and deadly stab maneuvers are not talked about or mentioned.
- I suggest you get the latest changes from dev by renaming your ForestZone scene to TutorialPoC then doing a "git merge dev" in GitBash.

Other than that, the tutorial is awesome! 👍

**Review by T-aut — APPROVED** [review]
Daaaamn this tutorial looks real good 💯. I like the new text, seems to fit the style. I partially agree with @MellowJay on the fact that special attacks are not mentioned in the tutorial. _But it may be the case_ that these attacks should be like a hidden thing - something that the player is taught later on (but skilled players know how to use from the start).

Also the dialog box could be a tad bit higher in the canvas :)

**Pull request merged**